homeassistant/components/workday/binary_sensor.py

```diff
@@ -165,17 +174,45 @@ async def async_setup_entry(
         _holiday_string = holiday_date.strftime("%Y-%m-%d")
         LOGGER.debug("%s %s", _holiday_string, name)
 
+    entity = IsWorkdaySensor(
+        obj_holidays,
+        workdays,
+        excludes,
+        days_offset,
+        sensor_name,
+        entry.entry_id,
+    )
+
+    async def check_date(service_call: ServiceCall) -> ServiceResponse:
+        """Check if date is workday or not."""
+        date_to_test: date = service_call.data[CHECK_DATE]
+        holiday_date = (
+            date_to_test in entity._obj_holidays  # pylint: disable=protected-access
```

Comment:
Out of curiosity, is there a reason to grab `entity._obj_holidays` instead of just using the `obj_holidays` variable directly? It seems like the latter would let you avoid the pylint suppression.

Reply:
The one belonging to the entity is adjusted by the users settings.

Reply:
I don't think I see anything that modifies `_obj_holidays` on the entity. If the user changes their settings, doesn't that just result in a new entity getting created and registered? (And presumably, a new copy of the service method getting created and registered as well)